src/osvr/Client/RenderManagerConfig.cpp

```diff
@@ -0,0 +1,190 @@
+/** @file
+    @brief OSVR rendermanager configuration
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com>
+
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/Client/RenderManagerConfig.h>
+#include <osvr/Common/ClientContext.h>
+#include <osvr/Util/Verbosity.h>
+
+// Library/third-party includes
+#include <json/value.h>
+#include <json/reader.h>
+
+namespace osvr {
+namespace client {
+
+    RenderManagerConfigPtr RenderManagerConfigFactory::create(OSVR_ClientContext ctx) {
+        try {
+            auto const configString = ctx->getStringParameter("/display");
+            RenderManagerConfigPtr cfg(new RenderManagerConfig(configString));
+            return cfg;
+        }
+        catch (std::exception const &e) {
+            OSVR_DEV_VERBOSE(
+                "Couldn't create a render manager config internally! Exception: "
+                << e.what());
+            return RenderManagerConfigPtr{};
+        }
+        catch (...) {
+            OSVR_DEV_VERBOSE("Couldn't create a display config internally! "
+                "Unknown exception!");
+            return RenderManagerConfigPtr{};
+        }
+    }
+
+    RenderManagerConfig::RenderManagerConfig(const std::string& render_manager_config)
+    {
+        parse(render_manager_config);
+    }
+
+    void RenderManagerConfig::parse(const std::string& render_manager_config)
+    {
+      Json::Reader reader;
+      Json::Value root;
+      reader.parse(render_manager_config, root, false);
```

Comment:
Could fail here, by returning false - there's a JsonHelper.h or something in common that wraps this in a friendly way (throwing an exception if it blows up, instead of an easy-to-ignore return code.)